test/tst_xtcpsocket.cpp

```diff
@@ -0,0 +1,57 @@
+#include <QFuture>
+#include <QtConcurrent/QtConcurrent>
+#include <QtTest/QTest>
+
+#include "io/XTcpSocket.h"
+
+class tst_xtcpsocket : public QObject {
+    Q_OBJECT
+
+private:
+    XTcpSocket client;
+    XTcpSocket server;
+
+private slots:
+    void tst_pinpong();
+};
+
+void tst_xtcpsocket::tst_pinpong() {
```

Comment:
Awesome that these tests are passing on all systems! 👍

Reply:
I believe we won't have any problems with sockets.